TESTING.md

```diff
@@ -0,0 +1,53 @@
+<!-- TOC -->
+- [Testing](#testing)
+    - [Code Coverage Collecting](#code-coverage-collecting)
+    - [Code Coverage Reporting](#code-coverage-reporting)
+<!-- TOC -->
+
+## Testing
+
+### Code Coverage Collecting
+Code coverage is a measurement of how many lines of code are executed under the test suites.
+
+#### OpenSearch plugin
+
+##### Gradle project
+1. Add [JaCoCo Gradle plugin](https://docs.gradle.org/current/userguide/jacoco_plugin.html) to `build.gradle` in your repository. [example](https://github.com/opensearch-project/sql/blob/opensearch-1.0.0.0-rc1/build.gradle#L48)
+2. Set Jacoco to export coverage report in `xml` format. [example](https://github.com/opensearch-project/sql/blob/opensearch-1.0.0.0-rc1/build.gradle#L101-L105)
+3. Call gradle task "jacocoTestReport" to generate report (`./gradlew jacocoTestReport`) after running your tests. (Or add a `dependsOn` to make "jacocoTestReport" task depends on your test tasks. [example](https://github.com/opensearch-project/index-management/blob/opensearch-1.0.0.0-beta1/build-tools/coverage.gradle))
+
+##### Maven project
+Since OpenSearch uses Gradle as the build tool, we don't recommend using Maven.
+Code coverage for a Maven project can be generated by [JaCoCo Maven plugin](https://www.eclemma.org/jacoco/trunk/doc/maven.html), and we also got an [example](https://github.com/opensearch-project/security/blob/v1.0.0.0-rc1/pom.xml#L636).
+
+#### OpenSearch Dashboards plugin
```

Comment:
Just add a TBD and then we could add this section later.

Reply:
I will write the example of using Jest as the testing framework and generate code coverage report, as the existing plugins are using, so that here will not be empty. 😁

Reply:
Yay!!

Reply:
It's very easy to get code coverage report when using Jest testing framework, and I added the guide.